File: code/jvm/im-http-api/src/main/kotlin/im/api/model/output/invitations/ChannelInvitationsPaginatedOutputModel.kt
package im.api.model.output.invitations

import im.api.model.output.PaginationOutputModel
import im.domain.invitations.ChannelInvitation
import im.repository.pagination.Pagination

data class ChannelInvitationsPaginatedOutputModel(
    val invitations: List<ChannelInvitationOutputModel>,
    val pagination: PaginationOutputModel? = null,
) {
    companion object {
        fun fromDomain(invitations: Pagination<ChannelInvitation>): ChannelInvitationsPaginatedOutputModel =
            ChannelInvitationsPaginatedOutputModel(
                invitations = invitations.items.map { ChannelInvitationOutputModel.fromDomain(it) },
                pagination = PaginationOutputModel.fromPagination(invitations.info),
            )
    }
}

File: code/client/web/src/Dto/output/invitations/ChannelInvitationsPaginatedOutputModel.ts
import { PaginationOutputModel } from '../PaginationOutputModel';
import { ChannelInvitationOutputModel } from './ChannelInvitationOutputModel';

/**
 * Represents the output model for a paginated list of channel invitations.
 *
 * @param invitations The list of channel invitations.
 * @param pagination The pagination information.
 */
export interface ChannelInvitationsPaginatedOutputModel {
    invitations: ChannelInvitationOutputModel[];
    pagination: PaginationOutputModel;
}
